src/services/EcoBridge/Xdai/XdaiBridge.reducer.ts

```diff
@@ -0,0 +1,136 @@
+import { createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit'
+import { TokenList } from '@uniswap/token-lists'
+
+import { BridgeTransactionStatus } from '../../../state/bridgeTransactions/types'
+import { BridgeDetails, BridgingDetailsErrorMessage, SyncState, XdaiBridgeList } from '../EcoBridge.types'
+import { xdaiBridgeTransactionAdapter } from './XdaiBridge.adapter'
+import { XdaiBridgeTransaction, XdaiMessage } from './XdaiBridge.types'
+
+interface XdaiBridgeState {
+  bridgingDetails: BridgeDetails
+  bridgingDetailsStatus: SyncState
+  lastMetadataCt: number
+  listsStatus: SyncState
+  lists: { [id: string]: TokenList }
+  transactions: EntityState<XdaiBridgeTransaction>
+}
+
+const initialState: XdaiBridgeState = {
+  transactions: xdaiBridgeTransactionAdapter.getInitialState({}),
+  bridgingDetails: {},
+  bridgingDetailsStatus: SyncState.IDLE,
+  lists: {},
+  listsStatus: SyncState.IDLE,
+  lastMetadataCt: 0,
+}
+
+const createXdaiSlice = (bridgeId: XdaiBridgeList) =>
+  createSlice({
+    name: bridgeId,
+    initialState,
+    reducers: {
+      setBridgeDetails: (state, action: PayloadAction<BridgeDetails>) => {
+        const { gas, fee, estimateTime, receiveAmount, requestId } = action.payload
+
+        //(store persist) crashing page without that code
+        if (!state.bridgingDetails) {
+          state.bridgingDetails = {}
+        }
+
+        if (requestId !== state.lastMetadataCt) {
```

Comment:
why are we doing this condition?

Reply:
https://github.com/levelkdev/swapr-dapp/pull/1089#discussion_r897876141